src/deprecated.js

```diff
@@ -79,9 +85,21 @@ jQuery.isNumeric = function( obj ) {
 		!isNaN( obj - parseFloat( obj ) );
 };
 
-jQuery.trim = function( text ) {
-	return text == null ?
-		"" :
-		( text + "" ).replace( rtrim, "" );
-};
+if ( String.prototype.trim ) {
+	useDefaultTrim = "_" === " \uFEFF\xA0_\uFEFF\xA0 ".trim();
+}
+
+if ( useDefaultTrim ) {
+	jQuery.trim = function( text ) {
+		return text == null ?
+			"" :
+			( text + "" ).trim();
+	};
+} else {
+	jQuery.trim = function( text ) {
+		return text == null ?
+			"" :
+			( text + "" ).replace( ltrim, "" ).replace( rtrim, "$1" );
+	};
+}
```

Comment:
This is pretty long and has repetitions. I'd incorporate the support check into the function to make it shorter; how about:
```suggestion
jQuery.trim = function( text ) {
	if ( text == null ) {
		return "";
	}
	return support.trim ?
		( text + "" ).trim() :
		( text + "" ).replace( ltrim, "" ).replace( rtrim, "$1" );
};

```
?